engine/src/main/java/com/redhat/cloud/notifications/events/EventConsumer.java

```diff
@@ -190,11 +214,47 @@ public CompletionStage<Void> process(Message<String> message) {
             processingExceptionCounter.increment();
             Log.infof(e, "Could not process the payload: %s", payload);
         } finally {
-            // bundleName[0] and appName[0] are null when the action parsing failed.
-            String bundle = bundleName[0] == null ? "" : bundleName[0];
-            String application = appName[0] == null ? "" : appName[0];
-            consumedTimer.stop(registry.timer(CONSUMED_TIMER_NAME, "bundle", bundle, "application", application));
+            consumedTimer.stop(registry.timer(
+                    CONSUMED_TIMER_NAME,
+                    TAG_KEY_BUNDLE, tags.getOrDefault(TAG_KEY_BUNDLE, ""),
+                    TAG_KEY_APPLICATION, tags.getOrDefault(TAG_KEY_APPLICATION, ""),
+                    TAG_KEY_EVENT_TYPE_FQN, tags.getOrDefault(TAG_KEY_EVENT_TYPE_FQN, "")
+            ));
         }
         return message.ack();
     }
+
+    private EventWrapper<?, ?> parsePayload(String payload, Map<String, String> tags) {
+        try {
+            Action action = actionParser.fromJsonString(payload);
+            tags.putIfAbsent(TAG_KEY_BUNDLE, action.getBundle());
+            tags.putIfAbsent(TAG_KEY_APPLICATION, action.getApplication());
+            return new EventWrapperAction(action);
+        } catch (RuntimeException actionParseException) {
```

Comment:
Can we catch something more specific than `RuntimeException`?

Reply:
Catching a custom `ParsingException` now.